### PR TITLE
Feat/shared module

### PR DESCRIPTION
**Pull request by Malak67**

### CODE REVIEW

File: services/explorer-api/src/modules/address/address.service.ts [hunk 1]
```diff
@@ -101,11 +102,13 @@ export class AddressService {
   }
 
   public async getBlocksProduced(address: string): Promise<BlocksCount> {
+    const parsedAddress = toChecksumAddress(address);
+
     const [{ pbfts_mined }] = await this.pbftRepository.query(
-      `SELECT COUNT(distinct hash) as pbfts_mined from pbfts WHERE LOWER(miner) = lower('${address}')`
+      `SELECT COUNT(distinct hash) as pbfts_mined from pbfts WHERE LOWER(miner) = lower('${parsedAddress}')`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:108]: If we search `lower` here the conversion to checksum is redundant

File: services/explorer-api/src/modules/address/address.service.ts [hunk 2]
```diff
@@ -101,11 +102,13 @@ export class AddressService {
...
   public async getBlocksProduced(address: string): Promise<BlocksCount> {
+    const parsedAddress = toChecksumAddress(address);
+
     const [{ pbfts_mined }] = await this.pbftRepository.query(
-      `SELECT COUNT(distinct hash) as pbfts_mined from pbfts WHERE LOWER(miner) = lower('${address}')`
+      `SELECT COUNT(distinct hash) as pbfts_mined from pbfts WHERE LOWER(miner) = lower('${parsedAddress}')`
     );
     const [{ dags_mined }] = await this.pbftRepository.query(
-      `SELECT COUNT(distinct hash) as dags_mined from dags  WHERE LOWER(author) = lower('${address}')`
+      `SELECT COUNT(distinct hash) as dags_mined from dags  WHERE LOWER(author) = lower('${parsedAddress}')`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:111]: same

File: services/explorer-api/src/modules/address/address.service.ts [hunk 3]
```diff
@@ -127,34 +130,38 @@ export class AddressService {
...
   }
 
   public async getTransactions(
     address: string
   ): Promise<TransactionResponse[]> {
+    const parsedAddress = toChecksumAddress(address);
     // Don't test with Swagger as it will break
     const res = await this.txRepository.query(
-      `SELECT t.hash, t.from, t.to, t.status, t.value, t."gasUsed", t."gasPrice", p.number as block, p.timestamp as age from transactions t LEFT JOIN pbfts p ON t."blockId" = p.id WHERE LOWER(t.from) = LOWER('${address}') or LOWER(t.to) = LOWER('${address}')`
+      `SELECT t.hash, t.from, t.to, t.status, t.value, t."gasUsed", t."gasPrice", p.number as block, p.timestamp as age from transactions t LEFT JOIN pbfts p ON t."blockId" = p.id WHERE LOWER(t.from) = LOWER('${parsedAddress}') or LOWER(t.to) = LOWER('${parsedAddress}')`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:144]: same redundancy

File: services/explorer-api/src/modules/address/address.service.ts [hunk 4]
```diff
@@ -127,34 +130,38 @@ export class AddressService {
...
-    if (!address) throw new BadRequestException('Address not supplied!');
+    if (!address) {
+      throw new BadRequestException('Address not supplied!');
+    }
+    const parsedAddress = toChecksumAddress(address);
     let balance = toBN('0');
     try {
       const [{ total_sent }] = await this.txRepository.query(
-        `select sum(value::REAL) as total_sent from transactions where lower(transactions.from) = lower('${address}');`
+        `select sum(value::REAL) as total_sent from transactions where lower(transactions.from) = lower('${parsedAddress}');`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:158]: same

File: services/explorer-api/src/modules/address/address.service.ts [hunk 5]
```diff
@@ -127,34 +130,38 @@ export class AddressService {
...
       );
       const [{ total_received }] = await this.txRepository.query(
-        `select sum(value::REAL) as total_received from transactions where lower(transactions.to) = lower('${address}');`
+        `select sum(value::REAL) as total_received from transactions where lower(transactions.to) = lower('${parsedAddress}');`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:161]: same

File: services/explorer-api/src/modules/address/address.service.ts [hunk 6]
```diff
@@ -127,34 +130,38 @@ export class AddressService {
...
       );
       const [{ total_mined }] = await this.txRepository.query(
-        `select sum(reward::REAL) as total_mined from pbfts where lower(miner) = lower('${address}');`
+        `select sum(reward::REAL) as total_mined from pbfts where lower(miner) = lower('${parsedAddress}');`
```
Comment (VargaElod23) [inline services/explorer-api/src/modules/address/address.service.ts:164]: same

**Review by VargaElod23 — APPROVED** [review]
Small redundancies to remove but LGTM 👍🏼

File: services/explorer-api/src/modules/address/address.service.ts (hunk 1 above)
Comment (Malak67) [reply services/explorer-api/src/modules/address/address.service.ts:108]: Yeah but checksum is doing something different, not changing them to lowercase. I would keep both just to be safe.

**Pull request merged**